install_deps_debug.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This script builds and installs additional debug, performance, and analysis tools for Concord-BFT development.
+# The script was validated on specific Linux distribution only (Ubuntu 18.04 LTS).
+# If you prefer working w/o docker at your dev station just run the script with sudo.
+# If you need to add any tool or dependency this is the right place to do it.
+# If you install temporary packages with apt-get and would like to remove them, add them to apt_pkgs_to_purge to
+# be removed at the end of the script.
+
+set -ex
+
+APT_GET_INSTALL_FLAGS="-y --no-install-recommends"
+TMPDIR=/tmp/
+apt_pkgs_to_purge=() # To save time between calls to installs, purge all packages only at the end
+
+# Install boost permanent and temporary library.
+# Currently, only temporary libraries are installed. These libraries are removed in uninstall_boost_temporary_libraries.
+BOOST_BOOTSTRAP_TEMPORARY_LIBRARIES="system,iostreams,filesystem" # comma-seperated
+install_boost() {
+    cd ${TMPDIR}
+    git clone --recurse-submodules --depth=1 --single-branch --branch=boost-1.80.0 https://github.com/boostorg/boost.git
```

Comment:
Why do you need a separate installation of boost for debug?

Reply:
These are additional (temporary) libraries not installed on release image. And this infra function allows to install permanent libraries easily in the future (need seperate string). These libraries are installed temporarily and then removed. They are needed for heaptrack build, and then removed.

Reply:
Is there a reason not to install a precompiled version? (This is a debug image after all..).
I think it makes more sense just installing one and not messing with dependencies.
In case this is not possible, please add a comment in above the install_boost function that explains that these boost libraries are not installed by the release image and are required for the compilation of heaptrack.

Reply:
The original build was with pre-compiled but with Toly changes to upgrade to boost 1.80, installing the full default boost 1.65 will take long time (then uninstall) , might break things in the image, and complicates the build and cmake commands. So I keep with what we have also to save time - install boost 1.80 which doesnt have precompiled header for ubuntu 18.04. If one look at the install_ deps.sh should understand this.
I'll add comments.

Reply:
I meant a precompiled heaptrack.

Reply:
heaptrack 1.0.0 is installed by default On ubunru 18.04
Too old version with many bugs and performance issues.
This one is 1.4.0